### PR TITLE
Add missing pieces for 'driver' support on macOS

### PR DESCRIPTION
**Pull request by stuartmorgan-g**
## Description

- Adds desktop projects to ApplicationPackageStore
- Plumbs target overrides through the desktop builds

## Related Issues

None

## Tests

I added the following tests: None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I've verified that this is enough to make flutter-driver work for macOS. I haven't tested Windows or Linux yet, but the corresponding pieces were obviously missing there as well so I added them at the same time.

File: packages/flutter_tools/lib/src/application_package.dart [hunk 1]
```diff
@@ -392,6 +392,9 @@ class ApplicationPackageStore {
   AndroidApk android;
   IOSApp iOS;
   FuchsiaApp fuchsia;
+  LinuxApp linux;
+  MacOSApp macOS;
+  WindowsApp windows;
 
   Future<ApplicationPackage> getPackageForPlatform(TargetPlatform platform) async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/application_package.dart]: We could a simple smoke test here by asserting that the returned value for a particular platform is non-null

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
I ended up having to touch more files; `lib/main.dart` was hard-coded in a few places in the tooling rather than using `path.join` which made things unhappy on Windows.

**Comment by stuartmorgan** [comment]
In the interest of unblocking the macOS support, so that people can start using it on at least one platform, I've narrowed the scope of this to just macOS and removed the path.join change. I'll do a follow-up with Linux and Windows.

**Pull request merged**